### PR TITLE
NMS-12538: add /ipinterfaces and /snmpinterfaces endpoints 

### PR DESCRIPTION
**Pull request by RangerRick**
This PR adds (read-only) `/ipinterfaces` and `/snmpinterfaces` endpoints under ReSTv2. (this is a rework of a previous PR after rearranging some things project-wise)

* JIRA (Issue Tracker): http://issues.opennms.org/browse/ULU-62
* Bamboo (Continuous Integration): https://bamboo.opennms.org/

### CODE REVIEW

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/SnmpInterfaceRestServiceIT.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import org.opennms.core.test.db.annotations.JUnitTemporaryDatabase;
+import org.opennms.core.test.rest.AbstractSpringJerseyRestTestCase;
+import org.opennms.test.JUnitConfigurationEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+/**
+ * TODO
```
Comment (j-white) [inline opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/SnmpInterfaceRestServiceIT.java]: Relevant?

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/IpInterfaceRestServiceIT.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.opennms.core.test.db.annotations.JUnitTemporaryDatabase;
+import org.opennms.core.test.rest.AbstractSpringJerseyRestTestCase;
+import org.opennms.test.JUnitConfigurationEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+/**
+ * TODO
```
Comment (j-white) [inline opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/IpInterfaceRestServiceIT.java]: Relevant?

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/SnmpInterfaceRestServiceIT.java (hunk 1 above)
Comment (RangerRick) [reply opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/SnmpInterfaceRestServiceIT.java]: LOL copy+paste from another rest interface

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/IpInterfaceRestServiceIT.java (hunk 2 above)
Comment (RangerRick) [reply opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/IpInterfaceRestServiceIT.java]: same

**Comment by RangerRick** [comment]
Added documentation.

**Pull request merged**